### PR TITLE
IN-229: new route /self.

### PR DESCRIPTION
**Pull request by n3op2**
### What

Adding a new route `/self` for retrieving `SELF_ADDRESS` env variable or `devDefault`

#### Changes
- api-doc update including route and request validator
- integration test along with other routes

```
  routes
    authenticated routes
      ✔ return membership members
      ✔ return membership members with aliases
      ✔ update non-existing member alias
      ✔ update existing member alias
      ✔ update existing member alias
      ✔ update alternative non-existing member with duplicate alias
      ✔ get member by alias
      ✔ get member by incorrect alias
      ✔ get member by invalid alias
      ✔ get member by address
      ✔ get self address or return default <--- THIS ONE
```

### CODE REVIEW

File: test/integration/routes.test.js [hunk 1]
```diff
@@ -13,6 +18,7 @@ const { AUTH_ISSUER, AUTH_AUDIENCE } = require('../../app/env')
 const { cleanup } = require('../seeds/members')
 
 describe('routes', function () {
+  this.timeout(3000)
```
Comment (n3op2) [inline test/integration/routes.test.js:21]: If anyone asks needed to increase as default is 1s and my PC is to slow :sad_parrot:

File: app/env.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ if (process.env.NODE_ENV === 'test') {
 const vars = envalid.cleanEnv(
   process.env,
   {
+    SELF_ADDRESS: envalid.str({ devDefault: '5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty' }),
```
Comment (mattdean-digicatapult) [inline app/env.js:15]: `SELF_ADDRESS` should be added to helm chart and to the readme

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
LGTM, just need to add the new env variable to the helm chart and readme

File: helm/dscp-identity-service/values.yaml [hunk 3]
```diff
@@ -5,6 +5,7 @@ config:
   dbName: dscp
   dbPort: 5432
   enableLivenessProbe: true
+  selfAddress: YYrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
```
Comment (dblane-digicatapult) [inline helm/dscp-identity-service/values.yaml:8]: What is this default value?

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: helm/dscp-identity-service/values.yaml (hunk 3 above)
Comment (n3op2) [reply helm/dscp-identity-service/values.yaml:8]: Just some random address

Comment (mattdean-digicatapult) [reply helm/dscp-identity-service/values.yaml:8]: It doesn't look valid. I thought all these identites started with a `5`? Anyway it should probably match the code default of `5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty`

File: test/integration/routes.test.js (hunk 1 above)
Comment (mattdean-digicatapult) [reply test/integration/routes.test.js:21]: this is ok. Yes sad but it's integration tests in blockchain so not overly surprising

File: helm/dscp-identity-service/values.yaml (hunk 3 above)
Comment (n3op2) [reply helm/dscp-identity-service/values.yaml:8]: it worked when I tested locally, updated to what you have suggested, same as in devDefault

**Review by mattdean-digicatapult — DISMISSED** [review]

**Review by mattdean-digicatapult — DISMISSED** [review]

**Review by dblane-digicatapult — DISMISSED** [review]

**Review by dblane-digicatapult — APPROVED** [review]

**Pull request merged**